src/main/java/io/vertx/core/file/CopyOptions.java

```diff
@@ -0,0 +1,162 @@
+   * Default constructor.
+   */
+  public CopyOptions() {
+  }
+
+  /**
+   * Copy constructor.
+   *
+   * @param other the options to copy
+   */
+  public CopyOptions(CopyOptions other) {
+    this.replaceExisting = other.replaceExisting;
+    this.copyAttributes = other.copyAttributes;
+    this.atomicMove = other.atomicMove;
+    this.nofollowLinks = other.nofollowLinks;
+  }
+
+
+  /**
+   * Constructor to create options from JSON.
+   *
+   * @param json the JSON
+   */
+  public CopyOptions(JsonObject json) {
+    this();
+    CopyOptionsConverter.fromJson(json, this);
+  }
+
+  /**
+   * @return true if an existing file, empty directory, or link should be replaced, false otherwise
+   */
+  public boolean isReplaceExisting() {
+    return replaceExisting;
+  }
+
+  /**
+   * Whether an existing file, empty directory, or link should be replaced. Defaults to {@code false}.
+   *
+   * @param replaceExisting true to replace, false otherwise
+   * @return a reference to this, so the API can be used fluently
+   */
+  public CopyOptions setReplaceExisting(boolean replaceExisting) {
+    this.replaceExisting = replaceExisting;
+    return this;
+  }
+
+  /**
+   * @return true if the file attributes should be copied, false otherwise
+   */
+  public boolean isCopyAttributes() {
+    return copyAttributes;
+  }
+
+  /**
+   * Whether the file attributes should be copied. Defaults to {@code false}.
+   *
+   * @param copyAttributes true to copy attributes, false otherwise
+   * @return a reference to this, so the API can be used fluently
+   */
+  public CopyOptions setCopyAttributes(boolean copyAttributes) {
+    this.copyAttributes = copyAttributes;
+    return this;
+  }
+
+  /**
+   * @return true if move should be performed as an atomic filesystem operation, false otherwise
+   */
+  public boolean isAtomicMove() {
+    return atomicMove;
+  }
+
+  /**
+   * Whether move should be performed as an atomic filesystem operation. Defaults to {@code false}.
+   *
+   * @param atomicMove true to perform as an atomic filesystem operation, false otherwise
+   * @return a reference to this, so the API can be used fluently
+   */
+  public CopyOptions setAtomicMove(boolean atomicMove) {
+    this.atomicMove = atomicMove;
+    return this;
+  }
+
+  /**
+   * @return true if the operation should not follow links, false otherwise
+   */
+  public boolean isNofollowLinks() {
+    return nofollowLinks;
+  }
+
+  /**
+   * Whether symbolic links should not be followed during copy or move operations. Defaults to {@code false}.
+   *
+   * @param nofollowLinks true to not follow links, false otherwise. Defaults to {@code false}.
+   * @return a reference to this, so the API can be used fluently
+   */
+  public CopyOptions setNofollowLinks(boolean nofollowLinks) {
+    this.nofollowLinks = nofollowLinks;
+    return this;
+  }
+
+  /**
+   * @return a set of corresponding {@link CopyOption}
+   */
+  @GenIgnore
+  public Set<CopyOption> toCopyOptionSet() {
```

Comment:
move this implementation detail to `FileSystem` as it's not really needed in the API